### PR TITLE
Expose lastUpdate and isIndexing in /stats endpoint

### LINKED ISSUE #3843: Metrics: expose two new indexing stats
Related product discussion: https://github.com/meilisearch/product/discussions/625#discussioncomment-6160845
Related spec: WIP

## Motivation

It would be nice to have some additional prometheus metrics that already exist in /stats endpoint:

    lastUpdate exposed as a gauge with seconds since last update, or unix timestamp to detect if something is not "feeding meilisearch with new data" so we can define alert rules to prevent stale data.
    isIndexing exposed as 0/1 would be helpful to correlate slower http_response_time_seconds_bucket with indexing

## Usage

To help work on this issue, we recently pushed a basic prometheus scraper + grafana dashboard that you can use (and that you should update with the new counters) in the `assets` directory of the repository.

## TODO

- [ ] Merge the PR into main
- [ ] Update the spec
- [ ] Update the product discussion

### PR DESCRIPTION
**Pull request by gentcys**
# Pull Request

## Related issue
Fixes #3843

## What does this PR do?
- expose lastUpdate in `/stats` endpoint
- expose isIndex in `stats` endpoint
- add a method `is_task_processing` in index-scheduler/src/lib.rs.

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: index-scheduler/src/lib.rs [hunk 1]
```diff
@@ -794,6 +794,12 @@ impl IndexScheduler {
         Ok(res)
     }
 
+    // Return true if there is at least one task that is processing.
+    pub fn is_task_processing(&self) -> Result<bool> {
+        let processing_tasks = self.processing_tasks.read().unwrap().processing.len();
+        Ok(processing_tasks > 0)
+    }
```
Comment (Kerollmops) [inline index-scheduler/src/lib.rs:801]: I like this new function, but you can use the `RoaringBitmap::is_empty` helper.
```suggestion
    pub fn is_task_processing(&self) -> Result<bool> {
        Ok(!self.processing_tasks.read().unwrap().processing.is_empty())
    }
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you very much for your PR 🥏

File: index-scheduler/src/lib.rs (hunk 1 above)
Comment (gentcys) [reply index-scheduler/src/lib.rs]: @Kerollmops 
Applied the suggestion in a new commit.

File: index-scheduler/src/lib.rs [hunk 2]
```diff
@@ -796,8 +796,7 @@ impl IndexScheduler {
 
     // Return true if there is at least one task that is processing.
     pub fn is_task_processing(&self) -> Result<bool> {
-        let processing_tasks = self.processing_tasks.read().unwrap().processing.len();
-        Ok(processing_tasks > 0)
+        Ok(self.processing_tasks.read().unwrap().processing.is_empty())
```
Comment (Kerollmops) [inline index-scheduler/src/lib.rs:799]:
```suggestion
        Ok(!self.processing_tasks.read().unwrap().processing.is_empty())
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Whoops, you did a small mistake here 🤭

File: index-scheduler/src/lib.rs (hunk 2 above)
Comment (gentcys) [reply index-scheduler/src/lib.rs:799]: Oh my...

Comment (gentcys) [reply index-scheduler/src/lib.rs]: Nice catch, fixed in a new commit.

**Review by Kerollmops — APPROVED** [review]
Thank you very much!
bors merge

**Comment by gentcys** [comment]
@Kerollmops 
It looks the test I added can't pass, could you guide me to fix it? I'm not familiar with snapshot test.

**Review by Kerollmops — APPROVED** [review]
Thank you very much!
Should be good!
bors merge

**Comment by dureuill** [comment]
bors merge

**Comment by curquiza** [comment]
Indeed, @dureuill looks like bors is stuck for a reason I don't know, let's rebase as you said, I don't see a better solution

**Review by dureuill — APPROVED** [review]
bors merge

**Comment by dureuill** [comment]
bors cancel

**Comment by dureuill** [comment]
bors merge

this time with the correct base branch!

**Review by dureuill — APPROVED** [review]

**Pull request merged**